contrib/pkg/clusterpool/clusterpool.go

```diff
@@ -0,0 +1,464 @@
+			o.Region = "centralus"
+		case cloudGCP:
+			o.Region = "us-east1"
+		}
+	}
+
+	if len(o.CloudSecretName) == 0 {
+		o.createCloudSecret = true
+	}
+
+	return nil
+}
+
+// validate ensures that option values make sense
+func (o *ClusterPoolOptions) validate(cmd *cobra.Command) error {
+	if !o.UseClusterImageSet && len(o.ClusterImageSet) > 0 {
+		cmd.Usage()
+		log.Info("If not using cluster image sets, do not specify the name of one")
+		return fmt.Errorf("invalid option")
+	}
+	if !validClouds[o.Cloud] {
+		cmd.Usage()
+		log.Infof("Unsupported cloud: %s", o.Cloud)
+		return fmt.Errorf("unsupported cloud: %s", o.Cloud)
+	}
+
+	return nil
+}
+
+// run executes the command
+func (o *ClusterPoolOptions) run() error {
+	if err := apis.AddToScheme(scheme.Scheme); err != nil {
+		return err
+	}
+
+	objs, err := o.generateObjects()
+	if err != nil {
+		return err
+	}
+
+	rh, err := utils.GetResourceHelper("create-pool")
+	if err != nil {
+		return err
+	}
+	if len(o.Namespace) == 0 {
+		o.Namespace, err = utils.DefaultNamespace()
+		if err != nil {
+			log.Error("Cannot determine default namespace")
+			return err
+		}
+	}
+	for _, obj := range objs {
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			log.WithError(err).Errorf("Cannot create accessor for object of type %T", obj)
+			return err
+		}
+		accessor.SetNamespace(o.Namespace)
+		if _, err := rh.ApplyRuntimeObject(obj, scheme.Scheme); err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func (o *ClusterPoolOptions) generateObjects() ([]runtime.Object, error) {
+	pullSecret, err := utils.GetPullSecret(o.PullSecret, o.PullSecretFile)
+	if err != nil {
+		return nil, err
+	}
+
+	if o.createCloudSecret {
+		switch o.Cloud {
+		case cloudAWS:
+			defaultCredsFilePath := filepath.Join(o.homeDir, ".aws", "credentials")
+			accessKeyID, secretAccessKey, err := awsutils.GetAWSCreds(o.CredsFile, defaultCredsFilePath)
+			if err != nil {
+				log.WithError(err).Error("Failed to get AWS credentials")
+				return nil, err
+			}
+			o.awsAccessKeyID = accessKeyID
+			o.awsSecretAccessKey = secretAccessKey
+		case cloudAzure:
+			creds, err := azureutils.GetCreds(o.CredsFile)
+			if err != nil {
+				log.WithError(err).Error("Failed to read in Azure credentials")
+				return nil, err
+			}
+			o.azureServicePrincipal = creds
+		case cloudGCP:
+			creds, err := gcputils.GetCreds(o.CredsFile)
+			if err != nil {
+				log.WithError(err).Error("Failed to get GCP credentials")
+				return nil, err
+			}
+			o.gcpServiceAccount = creds
+		}
+	}
+
+	imageSet, err := o.configureImages()
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := o.build()
+	if err != nil {
+		return nil, err
+	}
+
+	if imageSet != nil {
+		result = append(result, imageSet)
+	}
+	if len(pullSecret) != 0 {
+		result = append(result, o.generatePullSecret(pullSecret))
+	}
+
+	return result, nil
+}
+
+func (o *ClusterPoolOptions) configureImages() (*hivev1.ClusterImageSet, error) {
+	if len(o.ClusterImageSet) > 0 {
```

Comment:
Should it be an error if the user uses `--image-set` in addition to either `--release-image` or `--release-image-source`?